feature/feature.go

```diff
@@ -61,6 +61,13 @@ const (
 	//
 	// alpha: v1.5
 	MachineSetPreflightChecks featuregate.Feature = "MachineSetPreflightChecks"
+
+	// InfraMachineNameFromTemplate is a feature gate for the now deprecated way of naming infra machines.
+	// With this feature gate enabled, infra machines will be named based on the infra machine template.
+	// The new behavior is that they get the same name as the machine.
+	//
+	// Deprecated: v1.7.
```

Comment:
Not sure if this makes sense. I wanted to deprecate it right away since the behavior already changed. However, this means the linter complains that we are using it :sweat: